src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/ClusterAwareMetricReporter.java

```diff
@@ -0,0 +1,34 @@
+package com.mysql.cj.jdbc.ha.plugins.failover;
+
+import com.mysql.cj.log.Log;
+import com.mysql.cj.util.StringUtils;
+
+public class ClusterAwareMetricReporter {
+    private static ClusterAwareMetricContainer metricContainer = ClusterAwareMetricContainer.getInstance();
+
+    public static void reportMetrics(String connUrl, Log log, boolean atClusterLevel) {
+        if (StringUtils.isNullOrEmpty(connUrl) || log == null) {
+            return;
+        }
+        final ClusterAwareMetrics metrics = metricContainer.getOrCreate(connUrl, atClusterLevel);
+        StringBuilder logMessage = new StringBuilder(256);
+
+        logMessage.append("** Performance Metrics Report for '");
+        logMessage.append(connUrl);
+        logMessage.append("' **\n");
+        log.logInfo(logMessage);
+        metrics.reportMetrics(log);
+    }
+
+    public static void resetMetrics(String connUrl, boolean atClusterLevel) {
```

Comment:
I wouldn't be so granular in terms of what metrics to reset. I'd reset everything.

Reply:
This does reset all the metrics.